src/main/java/com/microsoft/azure/relay/WebSocketChannel.java

```diff
@@ -113,7 +113,17 @@ public void close() throws IOException {
 	public CompletableFuture<Void> writeAsync(ByteBuffer data, Duration timeout) {
 		return this.websocket.writeAsync(data, timeout);
 	}
-	
+	/**
+	 * Sends the text data to the remote endpoint within a timeout in one of the WriteModes.
+	 *
+	 * @param textData Text message to be sent.
+	 * @param timeout The timeout to connect to send the data within. May be null to indicate no timeout limit.
+	 * @return A CompletableFuture which completes when websocket finishes sending the data.
+	 * @throws TimeoutException Throws when the sending task does not complete within the given timeout.
+	 */
+	public CompletableFuture<Void> writeTextAsync(String textData, Duration timeout) {
```

Comment:
Just wondering, is there any functional difference where you would need to use WriteMode.TEXT instead?

Reply:
Perhaps if you want to control the encoding of the String. However, in our case there is no value in exposing the WriteMode or multi-part messages.